src/main/java/seedu/taskify/model/task/Date.java

```diff
@@ -46,6 +47,16 @@ public Date(LocalDateTime localDateTime) {
         this.value = localDateTime.toString();
     }
 
+    /**
+     * Check if a edited task's date is valid
+     *
+     */
+
+    public boolean isValidDateForEditing() {
+        LocalDateTime timeNow = LocalDateTime.now();
+        return this.localDateTime.isAfter(timeNow);
+    }
+
```

Comment:
Shouldnt this method be in `EditCommand` since Date shouldn't know anything about the details of whether date can be edited or not?